src/bot/notifier.ts

```diff
@@ -143,3 +143,104 @@ export function sendDeprecationMessage(
   return send(`⚠️ Deprecation warning:
 ${deprecationMessages[messageId]}`);
 }
+
+/**
+ * Request an OTP code from the user via Telegram and wait for their response
+ */
+export async function requestOtpCode(
+  companyId: string,
+  phoneNumber: string,
+): Promise<string> {
+  if (!bot || !telegramConfig?.chatId || !telegramConfig.enableOtp) {
+    throw new Error("Telegram OTP is not enabled or configured");
+  }
+
+  const message = await send(
+    `🔐 2FA Authentication Required\n\n` +
+      `Account: ${companyId}\n` +
+      `Please enter the OTP code sent to ${phoneNumber}:\n\n` +
+      `Reply to this message with the code (digits only).`,
+  );
+
+  if (!message) {
+    throw new Error("Failed to send OTP request message");
+  }
+
+  logger("Waiting for OTP code from user...");
+
+  return new Promise((resolve, reject) => {
```

Comment:
Can we chnage it to use a timeout abort signal? and maybe race 2 promises, one for the abort and one for the wait? i feel like this will result in nicer logic